package com.spring.controller;

import com.spring.model.Classe;
import com.spring.service.ClasseService;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;
import java.util.Optional;

/*
   File generated By
   Christooj_Generation_Code
*/
@RestController
@RequestMapping("/Classe")
@CrossOrigin("*")
public class ClasseController{
   @Autowired
   private ClasseService services;

   @GetMapping("/classeEcole")
   public List<Classe> getIdByRange(@RequestParam int startClasse, @RequestParam int endClasse) {
       return services.getIdBetween(startClasse, endClasse);
   }

   @PostMapping("/create")
    public String create(@RequestBody Classe object) {
    try {
        services.create(object);
        return "Cr��e avec succ�s";
    } catch (Exception e) {
        e.printStackTrace();
        return "Erreur lors de la cr�ation: " + e.getMessage();
    }
}

   @GetMapping("/find")
    public List<Classe> getAll() {
        try {
            List<Classe> all = services.read();
            return all;
        } catch (Exception e) {
            e.printStackTrace();
            return null;
        }
    }

   @GetMapping("/find/{id}")
    public Classe getById(@PathVariable int id) {
        try {
            Optional<Classe> optional = services.readById(id);
            return  optional.get();
        } catch (Exception e) {
            e.printStackTrace();
            return null;
        }
    }

   @PutMapping("/update/{id}")
    public Classe update(@PathVariable int id, @RequestBody Classe object) {
        try {
            Classe updated = services.update(id,object);
            return updated;
        } catch (Exception e) {
            e.printStackTrace();
            return null;
        }
    }

   @DeleteMapping("/delete/{id}")
    public String delete(@PathVariable int id) {
        try {
            services.delete(id);
            return "Success";
        } catch (Exception e) {
            e.printStackTrace();
           return "Il ya erreur: "+e.getMessage();
        }
    }


}

